components /HabitDetailsPage.js

```diff
@@ -0,0 +1,157 @@
+import styled from "styled-components";
+import Link from "next/link";
+
+const StyledHabitDetailsContainer = styled.section`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const StyledCategoriesContainer = styled.main`
+  margin: 0 1rem;
+  width: 90%;
+  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.2);
+  border-radius: 0.5rem;
+  background-color: beige;
+  padding: 1rem;
+`;
+
+const StyledHabitNameContainer = styled.header`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin: 6rem 1rem 1rem 1rem;
+  width: 80%;
+  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.2);
+  border-radius: 0.5rem;
+  background-color: beige;
+`;
+
+const HabitName = styled.h2`
+  font-size: 2rem;
+  margin: 1rem;
+`;
+
+const StyledHabitReasonsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin: 2rem 0 2rem;
+`;
+
+const StyledHabitReasonsLabel = styled.strong`
+  font-weight: bold;
+  padding: 0 0 0 1rem;
+`;
+
+const StyledHabitReasons = styled.p`
+  margin: 0;
+  border: 1px solid rgba(255, 235, 205, 0.5);
```

Comment:
Just a tiny tip: for reoccurring colors, i'd create a color variable in global styles